gnome-image-installer/pages/diskimage/gis-diskimage-page.c

```diff
@@ -117,6 +119,12 @@ gis_diskimage_page_selection_changed(GtkWidget *combo, GisPage *page)
   gis_store_set_image_signature (signature);
   g_free (signature);
 
+  if (checksum == NULL)
+    checksum = g_strjoin (NULL, image, ".sha256", NULL);
+
+  gis_store_set_image_checksum (checksum);
+  g_free (checksum);
```

Comment:
These days I would declare this variable as:

```
g_autofree char *checksum = NULL;
```

and then not need the manual `g_free`.

Reply:
Yes, that would be better. `signature` above is doing the same thing.